Add support for multiline PR comments

davidlopezre opened the pull request:
- Fix #1297

davidlopezre commented:
@googlebot I signed it!

dalzymendoza commented:
@googlebot I fixed it.

dalzymendoza commented:
@googlebot I consent.

gmlewis commented on github/github_test.go:
```diff
@@ -111,9 +111,14 @@ func testFormValues(t *testing.T, r *http.Request, values values) {
 func testHeader(t *testing.T, r *http.Request, header string, want string) {
-	if got := r.Header.Get(header); got != want {
-		t.Errorf("Header.Get(%q) returned %q, want %q", header, got, want)
+	relevantHeaders := r.Header[header]
```
Sorry, @davidlopezre - this change should be reverted, as it is not correct.  See below.

gmlewis commented on github/pulls_comments.go:
```diff
@@ -79,6 +81,7 @@ func (s *PullRequestsService) ListComments(ctx context.Context, owner string, re
 
 	// TODO: remove custom Accept header when this API fully launches.
 	req.Header.Set("Accept", mediaTypeReactionsPreview)
+	req.Header.Add("Accept", mediaTypeMultiLineCommentsPreview)
```
This is not how we set the Accept header when multiple fields are needed. Here is an example:

```
	// TODO: remove custom Accept header when this API fully launches.
	acceptHeaders := []string{mediaTypeLabelDescriptionSearchPreview, mediaTypeLockReasonPreview, mediaTypeDraftPreview}
	req.Header.Set("Accept", strings.Join(acceptHeaders, ", "))
```

gmlewis commented:
Reopening per #1297 conversation.

davidlopezre commented:
Hi @gmlewis,

Sorry for the delay in fixing this PR. I have addressed your comments.

Regarding the way we were adding the `Accept` headers, I have a question. Why is it preferable to concatenate them in a single key? I tried to look it up in the HTTP specification but I did not find anything related.

gmlewis commented:
> Regarding the way we were adding the Accept headers, I have a question. Why is it preferable to concatenate them in a single key? I tried to look it up in the HTTP specification but I did not find anything related.

All I can tell you is that we experimented several ways with the GitHub v3 API endpoints, and this mechanism is the only way we got it to work.

gmlewis commented:
Thank you, @wesleimp!
Merging.
